sljit_src/sljitNativePPC_common.c

```diff
@@ -652,20 +653,21 @@ SLJIT_API_FUNC_ATTRIBUTE void* sljit_generate_code(struct sljit_compiler *compil
 
 	compiler->error = SLJIT_ERR_COMPILED;
 	compiler->executable_offset = executable_offset;
-	compiler->executable_size = (sljit_uw)(code_ptr - code) * sizeof(sljit_ins);
 
 	code = (sljit_ins *)SLJIT_ADD_EXEC_OFFSET(code, executable_offset);
 
 #if (defined SLJIT_INDIRECT_CALL && SLJIT_INDIRECT_CALL)
 #if (defined SLJIT_CONFIG_PPC_64 && SLJIT_CONFIG_PPC_64)
-	if (((sljit_sw)code_ptr) & 0x4)
+	if ((sljit_sw)code_ptr & 0x4)
 		code_ptr++;
-#endif
+#endif /* SLJIT_CONFIG_PPC_64 */
 	sljit_set_function_context(NULL, (struct sljit_function_context*)code_ptr, (sljit_uw)code, (void*)sljit_generate_code);
```

Comment:
I feel this patch is over complicated. My suggestion is add the size to code_ptr in this case, and the computation moved below is correct. Then no other change is needed.

Reply:
While I agree it is more involved than needed and had indeed several bugs originally, I think the latest version is actually cleaner and simpler than the original.

Indeed it was the overloading of `code->size` part of the reason why it was far more complicated to understand and code than the s390x version, hence why I proposed that as (while at it) cleanup.

Reply:
Adding the context size after `sljit_set_function_context(NULL, (struct sljit_function_context*)code_ptr, (sljit_uw)code, (void*)sljit_generate_code);` is still missing. It is something like:

`code_ptr += sizeof (struct sljit_function_context) / sizeof(type pointed by code_ptr)`

This way the extra allocation at the end is also included in the size.

Reply:
A bit of background: function pointers on ppc may point to a descriptor structure, which contains the initial value of some global registers. These global values are not used by the jit compiler. However, I am not sure we correctly call functions from jit. We may not set these values as well. As long as the called function is in the same module, this is not a problem, but it can be when the application is big.

Reply:
> `code_ptr += sizeof (struct sljit_function_context) / sizeof(type pointed by code_ptr)`

we can't do this, because `code_ptr` must point to the beginning of the trampoline at the end.

> This way the extra allocation at the end is also included in the size.

now you see why I did the "cleanup"; this code is confusing and fragile and I suspect still buggy.